test/astar.cc

```diff
@@ -1617,6 +1631,120 @@ TEST(Astar, BiDirTrivial) {
   EXPECT_LT(path.front().elapsed_cost.secs, 1);
 }
 
+TEST(BiDiAstar, test_recost_path) {
```

Comment:
one other option here would have been to subclass the bidirectional a* class and then call form path with some preconstructed labels that use the shortcut. then you can always be sure that the test is actually confirming that FormPath is undoing any labels with shortcuts. dont worry about making this change unless you really want to, just an observation from me

Reply:
if you don't mind I will leave as is. But I completely agree, in unittests it's better to test smaller part than whole pipeline